src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -888,6 +891,121 @@ struct accel_deadlock_status
   }
 };
 
+// structure to get aie tiles(core, shim, mem) row start and num of rows 
+struct aie_tiles_row_info
+{
+  using result_type = boost::any;
+
+  static result_type
+  get(const xrt_core::device* dev,  key_type key, const boost::any& type)
+  {
+    auto tiles_info = asd_parser::get_aie_tiles_info(dev->get_user_handle());
+
+    auto tile_type = boost::any_cast<xrt_core::query::aie_tiles_row_info::tile_type>(type);
+
+    switch (tile_type) {
+      case xrt_core::query::aie_tiles_row_info::tile_type::core :
+        return std::make_tuple(tiles_info.cols, tiles_info.core_row_start, tiles_info.core_rows);
+      case xrt_core::query::aie_tiles_row_info::tile_type::shim :
+        return std::make_tuple(tiles_info.cols, tiles_info.shim_row_start, tiles_info.shim_rows);
+      case xrt_core::query::aie_tiles_row_info::tile_type::mem :
+        return std::make_tuple(tiles_info.cols, tiles_info.mem_row_start, tiles_info.mem_rows);
+      default :
+        throw std::runtime_error("Invalid tile type");
+    }
+  }
+}; 
+
+// common helper structure to get aie tiles(core, mem, shim) info from driver
+struct aie_status_helper
+{
+  static std::vector<char>
+  get_aie_cols_buf(const xrt_core::device* dev, uint32_t start_col, uint32_t num_cols, 
+      asd_parser::aie_tiles_info &info)
+  {
+    // do sanity checks
+    asd_parser::aie_info_sanity_check(start_col, num_cols, info);
+
+    // Get Aie column status from driver
+    std::vector<char> buf(info.col_size * num_cols);
+    asd_parser::get_aie_col_info(dev->get_user_handle(), buf.data(), info.col_size * num_cols, start_col, num_cols);
+
+    return buf;
+  }
+
+  template <typename tile_type>
+  static std::string
+  aie_info(const xrt_core::device* dev)
+  {
+    uint16_t major_ver, minor_ver;
+
+    // Get Aie status version and check compatibility
+    asd_parser::get_aie_status_version_info(dev->get_user_handle(), major_ver, minor_ver);
```

Comment:
```suggestion
   dev->device_query<xrt_core::query::aie_status>(major_ver, minor_ver);
    asd_parser::get_aie_status_version_info(dev->get_user_handle(), major_ver, minor_ver);
```
Going with my earlier comment in the header, we should be using the query infrastructure here.

This applies to the other `get_*` functions we have in this file